typing/printtyp.ml

```diff
@@ -1162,8 +1169,11 @@ let filter_params tyl =
     List.fold_left
       (fun tyl ty ->
         let ty = repr ty in
-        if List.memq ty tyl then Btype.newgenty (Tsubst ty) :: tyl
+        if List.memq ty tyl then Btype.newgenty (Ttuple [ty]) :: tyl
         else ty :: tyl)
+      (* Two parameters might be identical due to a constraint but we need to
+         print them differently in order to make the output syntactically valid.
+         We use Tsubst because it does not appear in a valid type. *)
```

Comment:
I am confused by this comment that says that `Tsubst` is used, but (if I read the diff correctly) in fact changes from `Tsubst` to `Ttuple`.

Reply:
Fixed.